app/src/main/java/com/sthoray/allright/ui/search/view/ResultsFragment.kt

```diff
@@ -42,6 +42,14 @@ class ResultsFragment : Fragment(R.layout.fragment_results) {
         setupFab()
         setListingOnClickListeners()
         setupObservers()
+        swipeToRefreshResultFragment()
+    }
+
+    private fun swipeToRefreshResultFragment() {
+        swipeRefreshResult.setOnRefreshListener {
+            viewModel.applyFiltersAndSearch()
+            swipeRefreshResult.isRefreshing = false
```

Comment:
As before, this causes the swipe to refresh spinner is cancelled as soon as it is started.

Reply:
The reason line 51 here is set  to false is to make sure that spinner is cancelled after we get the updated data. If we don't do that then the spinner will spin forever even after the data has been updated.